src/libdredd/src/mutate_ast_consumer.cc

```diff
@@ -90,6 +96,37 @@ void MutateAstConsumer::HandleTranslationUnit(clang::ASTContext& ast_context) {
     return;
   }
 
+  // Rewrite the size expressions of constant-sized arrays as needed.
+  for (const auto& constant_sized_array_decl :
+       visitor_->GetConstantSizedArraysToRewrite()) {
+    assert(compiler_instance_->getLangOpts().CPlusPlus &&
+           constant_sized_array_decl->getType()->isConstantArrayType());
+    auto constant_array_typeloc = constant_sized_array_decl->getTypeSourceInfo()
+                                      ->getTypeLoc()
+                                      .getAs<clang::ConstantArrayTypeLoc>();
+    if (constant_array_typeloc.isNull()) {
+      // In some cases a declaration with constant array type does not have
```

Comment:
I don't fully get this. Can you provide an example?

Reply:
Rephrased to:

```
      // In some cases a declaration with constant array type does not have
      // an associated ConstantArrayTypeLoc object. This happens, for example,
      // when structured bindings are used. For example, consider:
      //
      //     auto [x, y] = a;
      //
      // This yields a constant-sized array, but there is no explicit array type
      // declaration. In such cases, no action is required.
```